src/github.com/getlantern/go-update/update.go

```diff
@@ -325,8 +326,8 @@ func (u *Update) FromStream(updateWith io.Reader) (err error, errRecover error)
 			return
 		}
 	case PATCHTYPE_NONE:
```

Comment:
We weren't decoding the compressed bzip2 binary so the checksum verification always failed. This happened when the client checksum did not match with the binary that was published in the releases page.